source/codegen/service_helpers.py

```diff
@@ -174,6 +182,15 @@ def expand_varargs_parameters(parameters):
     return new_parameters
 
 
+def create_library_argument(parameter):
+    parameter_type = parameter.get('type', '')
+    # This is a hack. niRFmxInstr.h typdefs int8 as "char" (instead of "signed char"), but our
```

Comment:
Hopefully the comment explains this well enough - just calling it out in the PR.

Reply:
What happens if we change the `signed char[]` to `int8[]` in the extraction code? Does that just work?

I'm interpreting this as "2 wrongs":
1. CppHeaderParser doesn't preserve the nominal types for those semi-standard int types (in this case `int8`).
2. rfmx defines `int8` in a weird way.

For DAQ, we already compensate for (1) by correcting those types during extraction (for `int32` and `uint32`. So it seems reasonable to do that for `int8`. So `type` always matches the library type. But ... maybe it would still need to do something in the service/coercion code.

Reply:
Yeah, I was able to do this in scrapigen and I think it looks a bit cleaner here.